### ISSUE TITLE
Allow preview to not fail when protected resources are deleted

### ISSUE THREAD

**Issue by resentfulpancake** [pulumi/pulumi#18103]
`pulumi preview` _always_ fails when protected resources are removed from the code, e.g.:

```sh
$ pulumi preview
Previewing update (dev):
     Type                  Name        Plan     Info
     pulumi:pulumi:Stack   infra-dev            1 error
 -   ├─ …
 -   └─ aws:iam:Role       myRole      delete   1 error

Diagnostics:
  pulumi:pulumi:Stack (infra-dev):
    error: preview failed

  aws:iam:Role (myRole):
    error: Preview failed: resource "urn:pulumi:dev::infra::aws:iam/role:Role::myRole" cannot be deleted
    because it is protected. To unprotect the resource, either remove the `protect` flag from the resource in your Pulumi program and run `pulumi up`, or use the command:
    `pulumi state unprotect 'urn:pulumi:dev::infra::aws:iam/role:Role::myRole'`
```

While protection is cool and all during update, this is a pain in CICD pipelines or when I only need to check the effects of my changes without impacting the current state.  
I cannot currently find a way to ignore the protection and still execute the preview.

I would like to be able to preview changes even when resources are protected, maybe with a dedicated option and an information message by the protected resource side, e.g.:

```sh
$ pulumi preview --warn-protected
Previewing update (dev):
     Type                  Name        Plan     Info
     pulumi:pulumi:Stack   infra-dev            1 warning
 -   ├─ …
 -   └─ aws:iam:Role       myRole      delete   [protected]

Diagnostics:
  aws:iam:Role (myRole):
    warning: resource "urn:pulumi:dev::infra::aws:iam/role:Role::myRole" cannot be deleted
    because it is protected. To unprotect the resource, either remove the `protect` flag from the resource in your Pulumi program and run `pulumi up`, or use the command:
    `pulumi state unprotect 'urn:pulumi:dev::infra::aws:iam/role:Role::myRole'`

Resources:
    - 4 to delete
    473 unchanged
```

## Affected feature

Would it be possible to equip the `preview` command with an option to only warn about protected resources and not fail?

This in my opinion should be the default behavior, but even just the ability to not fail on preview would help a lot.

**Comment by ringods** [comment]
Related to #12195

**Comment by brandonpollack23** [comment]
So we have this flag already "continueOnError" we should add this case to that so previews continue on these errors

**Comment by lunaris** [comment]
Note: we should expose this flag in the automation API too

**Comment by heyzling** [comment]
Huge pain in big stacks when you need to delete many protected reosources. The problem is that every preview run does not show every protected resource at once and you have to go through many-many launches of `pulumi preview` to get and uprotect every resource marked for deletion.

So flag like `--continue-on-errors` would be very useful.

**Comment by benglewis** [comment]
@Frassle Unless I am missing something this doesn't actually address the original issue insofar as there is still no flag to allow ignoring protected resources and making `pulumi preview` actually return `0` in the cases where the only issues are protected resources. It is great that `pulumi preview` now always prints the resources summaries on preview, even when protected resources block the actual application of `pulumi up`, but AFAIK, this still means that there is no straightforward way to have a CI implementation which returns non-zero status code for Pulumi code which would fail entirely at `pulumi up` but still return zero if the only issue is protected resources which could easily be unprotected and then deleted and the `pulumi up` would succeed

**Comment by Frassle** [comment]
@benglewis we discussed it internally and felt that preview should error if up would error. When we discussed this internally we ended up a little hesitant to making further changes here. 

We could maybe do something like --ignore-protected-deletes or --warn-protect, but we'd want to think about how that should apply to both preview and up.

**Comment by benglewis** [comment]
@Frassle I agree that it makes sense that `up` and `preview` are consistent, but I don't understand why the ticket was closed when the feature was not implemented. If you are against creating the feature, that is fine - rejected issues are an option - it would just be great to have that communicated. If not, then it would be great if this issue could be re-opened

**Comment by Frassle** [comment]
No that's reasonable. We closed this on the basis that we had made it so you could preview in the presence of protect errors, all the summary information is now written out. But it still _errors_ rather than returns exit code 0.

As above, I'm not sure what the right thing to do here is, but the request of "I want to ignore protect delete errors for previews" seems reasonable. As opposed to the comment in the issue I doubt we'd ever make that the default behaviour, but it could be an option.

**Comment by tyrone-statsig** [comment]
Lets say I wanted a full list of URNS to unprotect, can I get the list for this using the CLI?
